### ISSUE TITLE
`load_crucible_llvm_module` uses implicit hidden state

### ISSUE THREAD

**Issue by TomMD** [GaloisInc/saw-script#206]
The new llvm load function has a type:

```
load_crucible_llvm_module : String -> TopLevel ()
```

Vs the prior function of:

```
llvm_load_module : String -> TopLevel LLVMModule
```

So in the past we had a style I'd say is a bit more flexible (supports more than one module at a time) and with more honest types since they aren't hiding effectful actions in some hazily defined top level monad.  Are there plans to change the crucible API?

**Comment by atomb** [comment]
I'd like to use a type more like the original `llvm_load_module`, too. There may be some implicit state in Crucible that made that awkward, but if not I'd like to make this change. I wonder if @robdockins knows why `load_crucible_llvm_module` implicitly sets a global module instead of returning a handle to one.

**Comment by robdockins** [comment]
@atomb, I don't think there is any principled reason.  I found it mildly easier when first hacking things together to do it that way.

**Comment by brianhuffman** [comment]
Another benefit of `llvm_load_module` is that it gives you a module value that you can `print`. When using `crucible_llvm_verify`, I have often started my saw-script files like this:
```
m <- llvm_load_module "foo.bc";
print m;
load_crucible_llvm_module "foo.bc";
```
which is just silly.

**Comment by brianhuffman** [comment]
This was fixed in revision b0feedfe76c55417ac898feab7b906b54c8d1ec6.
